src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/taglets/SnippetTaglet.java

```diff
@@ -501,13 +487,15 @@ private static String stringValueOf(AttributeTree at) throws BadSnippetException
     }
 
     private String languageFromFileName(String fileName) {
-        // TODO: find a way to extend/customize the list of recognized file name extensions
-        if (fileName.endsWith(".java")) {
-            return "java";
-        } else if (fileName.endsWith(".properties")) {
-            return "properties";
+        // The assumption is simple: a file extension is the language
+        // identifier.
+        // Was about to use Path.getExtension introduced in 8057113, but then
+        // learned that it was removed in 8298303.
+        int lastPeriod = fileName.lastIndexOf('.');
+        if (lastPeriod <= 0) {
+            return null;
         }
-        return null;
+        return (lastPeriod == fileName.length() - 1) ? null : fileName.substring(lastPeriod + 1);
```

Comment:
Some files, like `.gitignore`, only has suffixes, yet they are valid languages. What do you think?

Reply:
Sure, that's why there's `<= 0` and not `< 0` one line above that:

    int lastPeriod = fileName.lastIndexOf('.');
    if (lastPeriod <= 0) {
        return null;

So, if `fileName` starts with `.`, the extension is null.

Reply:
I think Chen's point was that it should not be `null`, to which I agree, although it is probably a corner case. 

One thing I would consider is converting the result or this method to lower case, since that is how the language attribute value is usually represented.

Reply:
> I think Chen's point was that it should not be `null`, to which I agree, although it is probably a corner case.

If so, my bad for misreading it. For such corner cases authors should specify `lang` attribute.

Reply:
General comment: while the dust on `Path.getExtension` is still settling, as a matter of policy, we should try and stay compatible with that. In particular, a filename like `.gitignore` is a hidden filename with no extension, not an empty filename with a 9-letter extension.

Reply:
Agreed on lower case

Reply:
> One thing I would consider is converting the result or this method to lower case, since that is how the language attribute value is usually represented.

> Agreed on lower case

Please provide a few examples where it would be necessary.